splash/tests/test_execute.py

```diff
@@ -1809,3 +1811,160 @@ def test_get_perf_stats(self):
+                          'outer': defaults.WINDOW_SIZE,
+                          'client': defaults.WINDOW_SIZE})
+
+    def test_window_size(self):
+        out = self.run_dims_test("""
+        function alter_state(splash)
+            splash:set_window_size("2000x2000")
+        end
+        """)
+        self.assertEqual(out,
+                         {'inner': '2000x2000',
+                          'outer': '2000x2000',
+                          'client': '2000x2000'})
+
+    def test_viewport_size(self):
+        out = self.run_dims_test("""
+        function alter_state(splash)
+            splash:set_viewport("2000x2000")
+        end
+        """)
+        self.assertEqual(out,
+                         {'inner': '2000x2000',
+                          'outer': defaults.WINDOW_SIZE,
+                          'client': '2000x2000'})
+
+    def test_viewport_overwrites_client_size(self):
+        out = self.run_dims_test("""
+        function alter_state(splash)
+            splash:set_window_size("1000x1000")
+            splash:set_viewport("2000x2000")
+        end
+        """)
+        self.assertEqual(out,
+                         {'inner': '2000x2000',
+                          'outer': '1000x1000',
+                          'client': '2000x2000'})
+
+    def test_window_size_overwrites_client_size(self):
+        out = self.run_dims_test("""
+        function alter_state(splash)
+            splash:set_viewport("2000x2000")
+            splash:set_window_size("1000x1000")
+        end
+        """)
+        self.assertEqual(out['inner'], '1000x1000')
+        self.assertEqual(out['outer'], '1000x1000')
+        self.assertEqual(out['client'], '1000x1000')
+
+    def test_window_size_validation(self):
+        cases = [
+            ('', 'Invalid window size format: '),
+            ('foobar', 'Invalid window size format: foobar'),
+            ('full', 'Invalid window size format: full'),
+            ('123', 'Invalid window size format: 123'),
+            ('123x', 'Invalid window size format: 123x'),
+            ('x', 'Invalid window size format: x'),
+            ('x123', 'Invalid window size format: x123'),
+            ('0x123', 'Window size is out of range'),
+            ('-1x123', 'Window size is out of range'),
+            ('99999x123', 'Window size is out of range'),
+            ('123x0', 'Window size is out of range'),
+            ('123x-1', 'Window size is out of range'),
+            ('123x99999', 'Window size is out of range'), ]
+
+        def run_test(size_str):
+            self.run_dims_test("""
+            function alter_state(splash)
+                splash:set_window_size(%s)
+            end
+            """ % (repr(size_str), ))
+
+        for size_str, errmsg in cases:
+            self.assertRaisesRegexp(RuntimeError, errmsg,
+                                    run_test, size_str)
+
+    def test_viewport_validation(self):
+        cases = [
```

Comment:
does not work: see the last paragraph of http://pytest.org/latest/unittest.html.